### ISSUE TITLE
Testing a LightGBM model on fresh data

### ISSUE THREAD

**Issue by j7zAhU** [microsoft/LightGBM#6044]
Apologies if I have posted this in the wrong section. This is more of a conceptual question. I performed combinatorial CV on 20 folds (5 choose 2, times 2) using early stopping. There are two labels and the positive class of interest is about 25% of the sample).  The model is very promising. 

I now have to prove this model to management using out of sample data. What is the recommended procedure here to determine the polished final model?

(1) Train on all 5 folds (how do I avoid overfitting with no early stopping set...)
(2) Choose one of the 20 boosters from the CV exercise at random
(3) Use all 20 boosters and go with the majority decision?
(4) Other..?

Many thanks

**Comment by jameslamb** [comment]
As a general rule, cross validation is not intended to produce a final model that's used for prediction in your application.

It's used to:

* choose between competing sets of hyperparameters
* get an estimate of the generalization error (how well the model will perform on unseen data)

Given only the details you've provided, I recommend that you:

* reserve a portion of your total dataset as a validation set (`X_valid`)
* perform cross-validation on the remaining data (`X_train`), to choose the best hyperparameters
* retrain on all of (`X_train`), using those chosen hyperparameters
    - use early stopping, and pass `X_valid` as the validation set

This is not really a LightGBM-specific question, so if you have additional not-specific-to-LightGBM questions, please try the following:

* review `scikit-learn`'s excellent tutorial on cross-validation: https://scikit-learn.org/stable/modules/cross_validation.html
* search https://stats.stackexchange.com/, e.g. review some of these answers:
*

**Comment by j7zAhU** [comment]
Thank you, I appreciate your help. Apologies for the generic question
